src/document_basic.c

```diff
@@ -1,94 +1,181 @@
 #include "document.h"
 #include "stemmer.h"
 #include "rmalloc.h"
+#include "module.h"
 
-void Document_Init(Document *doc, RedisModuleString *docKey, double score, int numFields,
-                   const char *lang, const char *payload, size_t payloadSize) {
+void Document_Init(Document *doc, RedisModuleString *docKey, double score, const char *lang) {
   doc->docKey = docKey;
   doc->score = (float)score;
-  doc->numFields = numFields;
-  doc->fields = numFields ? calloc(doc->numFields, sizeof(DocumentField)) : NULL;
-  doc->language = lang;
-  doc->payload = payload;
-  doc->payloadSize = payloadSize;
+  doc->numFields = 0;
+  doc->language = lang ? lang : DEFAULT_LANGUAGE;
+  doc->payload = NULL;
+  doc->payloadSize = 0;
 }
 
 void Document_AddField(Document *d, const char *fieldname, RedisModuleString *fieldval,
                        uint32_t typemask) {
   d->fields = realloc(d->fields, (++d->numFields) * sizeof(*d->fields));
   DocumentField *f = d->fields + d->numFields - 1;
   f->indexAs = typemask;
-  f->name = strdup(fieldname);  // free_detached called on this later on..
+  f->name = fieldname;
```

Comment:
Why not duplicate? If the api state const var then its better to duplicate and not take ownership.